File: actionbar/src/mobi/cangol/mobile/actionbar/view/ActionTabView.java
package mobi.cangol.mobile.actionbar.view;

import java.util.ArrayList;
import java.util.List;

import mobi.cangol.mobile.R;
import mobi.cangol.mobile.actionbar.ActionTab;
import mobi.cangol.mobile.actionbar.ActionTabItem;
import android.content.Context;
import android.util.AttributeSet;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.RadioGroup.OnCheckedChangeListener;

public class ActionTabView extends RadioGroup implements OnCheckedChangeListener {
	private LayoutInflater mInflater;
	private ActionTab mActionTab;
	private OnTabCheckedListener mOnTabCheckedListener;

	public ActionTabView(Context context, AttributeSet attrs) {
		super(context, attrs);
		this.setOrientation(HORIZONTAL);
		mInflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
		this.setOnCheckedChangeListener(this);
	}

	public void setActionTab(ActionTab actionTab) {
		this.mActionTab = actionTab;
	}

	public void addTabs(List<ActionTabItem> tabs) {
		for (int i = 0; i < tabs.size(); i++) {
			addTabItem(tabs.get(i));
		}
	}

	public void addTabItem(ActionTabItem tabItem) {
		this.addView(inflateTabItem(tabItem));
	}

	public void removeAllTabs() {
		this.removeAllViews();
	}

	private View inflateTabItem(ActionTabItem tabItem) {
		View view = mInflater.inflate(R.layout.actionbar_tab_item, this, false);

		RadioButton labelView = (RadioButton) view.findViewById(R.id.actionbar_tab_item);
		labelView.setId(tabItem.getId());
		labelView.setTag(tabItem);
		labelView.setText(tabItem.getTitle());
		labelView.setChecked(1 == tabItem.getChecked());
		return view;
	}

	public void setOnTabCheckedListener(OnTabCheckedListener onTabCheckedListener) {
		this.mOnTabCheckedListener = onTabCheckedListener;
	}

	public interface OnTabCheckedListener {
		boolean onTabChecked(ActionTabItem tab);
	}

	@Override
	public void onCheckedChanged(RadioGroup group, int checkedId) {
		View view = this.findViewById(checkedId);
		if (view == null)
			return;
		Log.d(VIEW_LOG_TAG, "first view=null");
		if (view.getTag() instanceof ActionTabItem) {
			final ActionTabItem tab = (ActionTabItem) view.getTag();
			if (mOnTabCheckedListener != null)
				mOnTabCheckedListener.onTabChecked(tab);
		}
	}

	public ArrayList<ActionTabItem> getTabs() {
		return mActionTab.getTabs();
	}

}

File: actionbar/src/main/java/mobi/cangol/mobile/actionbar/view/ActionTabView.kt
package mobi.cangol.mobile.actionbar.view

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.View
import android.widget.LinearLayout
import android.widget.RadioButton
import android.widget.RadioGroup
import android.widget.RadioGroup.OnCheckedChangeListener
import mobi.cangol.mobile.actionbar.ActionTab
import mobi.cangol.mobile.actionbar.ActionTabItem
import mobi.cangol.mobile.actionbar.R

/**
 * @author Cangol
 */
class ActionTabView : RadioGroup, OnCheckedChangeListener {
    private var mInflater: LayoutInflater? = null
    private var mActionTab: ActionTab? = null
    private var mOnTabSelectedListener: OnTabSelectedListener? = null

    val tabs: List<ActionTabItem>
        get() = mActionTab!!.tabs

    var tabSelected: Int
        get() = this.checkedRadioButtonId
        set(id) = this.check(id)

    constructor(context: Context) : super(context) {
        initViews(context)
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {
        initViews(context)
    }

    private fun initViews(context: Context) {
        this.orientation = LinearLayout.HORIZONTAL
        mInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        this.setOnCheckedChangeListener(this)
    }

    fun setActionTab(actionTab: ActionTab) {
        this.mActionTab = actionTab
    }

    fun addTabs(tabs: List<ActionTabItem>) {
        for (tabItem in tabs) {
            addTabItem(tabItem)
        }
    }

    fun addTabItem(tabItem: ActionTabItem) {
        this.addView(inflateTabItem(tabItem))
    }

    fun removeAllTabs() {
        this.removeAllViews()
    }

    private fun inflateTabItem(tabItem: ActionTabItem): View {
        val view = mInflater?.inflate(R.layout.actionbar_tab_item, this, false)

        val labelView = view?.findViewById(R.id.actionbar_tab_item) as RadioButton
        labelView.id = tabItem.id
        labelView.text = tabItem.text
        labelView.isChecked = 1 == tabItem.selected
        labelView.tag = tabItem
        return view
    }

    fun setOnTabSelectedListener(onTabSelectedListener: OnTabSelectedListener) {
        this.mOnTabSelectedListener = onTabSelectedListener
    }
    fun removeTabSelectedListener() {
        this.mOnTabSelectedListener = null
    }
    override fun onCheckedChanged(group: RadioGroup, checkedId: Int) {
        val view = this.findViewById<View>(checkedId) ?: return
        if (view.tag is ActionTabItem) {
           mOnTabSelectedListener?.onTabSelected(view.tag as ActionTabItem)
        }
    }

    interface OnTabSelectedListener {
        fun onTabSelected(tab: ActionTabItem): Boolean
    }

}
